bftengine/src/bftengine/messages/InternalMessage.hpp

```diff
@@ -63,6 +64,10 @@ using InternalMessage = std::variant<FullCommitProofMsg*,
                                      // Carries PrePrepare message after validation.
                                      PrePrepareCarrierInternalMsg,
 
+                                     // Add Carriers of messages which will translate
+                                     // Incoming External messages to Internal message.
+                                     CarrierMesssage*,
```

Comment:
What '*' does here?

Reply:
CarrierMessage just know the type of the message whose encapsulation it's carrying. So I am using a Pointer Type of CarrierMessage. This means, the internal queue will point to a location which will contain an encapsulated message with the type.